segment_store/src/segment.rs

```diff
@@ -139,6 +139,13 @@ impl Segment {
         &self.columns[*self.all_columns_by_name.get(name).unwrap()]
     }
 
+    // Returns a reference to a column's name and a column from the column name. The returned
```

Comment:
```suggestion
    // Returns a reference to specified column's name and the column itself. The returned
```

I found the use of the multiple use of column name (for input argument and output) to be slightly confusing. Maybe this is better?

Reply:
Not to get all inception and suggest a change to the suggestion but in my head I read it as `Returns a reference to *the* specified column's name and the column itself`? :-)

Reply:
How about:

```
// Takes a `ColumnName`, looks up that column in the `Segment`, and returns a reference to 
// that column's name owned by the `Segment` along with a reference to the column itself. 
// The returned column name will have the lifetime of `self`, not the lifetime of the input.

Reply:
lol
![S5u](https://user-images.githubusercontent.com/490673/101822610-91a86f00-3af7-11eb-854f-f802b098a39f.gif)